src/net.h

```diff
@@ -269,7 +269,7 @@ class CConnman
      * A non-malicious call (from RPC or a peer with addr permission) should
      * call the function without a parameter to avoid using the cache.
      */
-    std::vector<CAddress> GetAddresses(Network requestor_network, size_t max_addresses, size_t max_pct);
+    const std::vector<CAddress>& GetAddresses(CNode& requestor, size_t max_addresses, size_t max_pct);
```

Comment:
Changing the return type to a const reference doesn't actually make any difference here. The caller assigns a local variable `vAddr` to this function's return, so the vector needs to be copied in the calling function. Compare that to the current behavior of returning by value, where the vector is copied inside this function, but can be move-assigned in the caller.

This _would_ make a difference if we directly iterated over the return value, eg:

```
for (const CAddress &addr : connman.GetAddresses(pfrom, MAX_ADDR_TO_SEND, MAX_PCT_ADDR_TO_SEND)) {
    ...
```

or assigned the return of this function to a const reference and then iterated over it:

```
const std::vector<CAddress>& vAddr{connman.GetAddresses(pfrom, MAX_ADDR_TO_SEND, MAX_PCT_ADDR_TO_SEND)};
FastRandomContext insecure_rand;
for (const CAddress &addr : vAddr) {
    ...
```

but because `vAddr` can come from one of two places, I don't think it's worth doing that. Let's just keep this as return by value for simplicity.

Reply:
Thanks for the thorough explanation :)
I'm fine with keeping this as return by value for simplicity.